src/Atc.Rest.Client/EndpointResult.cs

```diff
@@ -0,0 +1,55 @@
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+
+namespace Atc.Rest.Client
+{
+    public class EndpointResult<TResponseContent> : EndpointResponse
+    {
+        public EndpointResult(EndpointResponse response)
+            : base(response)
+        {
+        }
+
+        [SuppressMessage("Major Code Smell", "S2372:Exceptions should not be thrown from property getters", Justification = "OK.")]
+        [SuppressMessage("Design", "CA1065:Do not raise exceptions in unexpected locations", Justification = "OK.")]
+        public TResponseContent SuccessContent
+        {
+            get
+            {
+                if (IsSuccess)
+                {
+                    return ContentObject switch
+                    {
+                        TResponseContent content => content,
+                        null => throw new InvalidDataException("Underlying ContentObject is null"),
+                        _ => throw new InvalidDataException($"Type is {ContentObject.GetType()}, but expected {typeof(TResponseContent)}")
+                    };
+                }
+
+                throw new InvalidDataException("The state is not success");
+            }
+        }
```

Comment:
```suggestion
        [return: MaybeNull]
        public TResponseContent SuccessContent
        {
            get
            {
                if (IsSuccess && ContentObject is TResponseContent content)
                {
                    return content;
                }
                
                return null;
            }
        }
```

I do not think that this should throw. Instead, it should return null when there is no content or no success. Its up to the user to check IsSuccess, just like with ErrorContent.